admin/src/components/dashboard.js

```diff
@@ -0,0 +1,220 @@
+import React, { Component } from 'react';
+
+import BarList from './barList';
+import BarDetails from './barDetails';
+
+class Dashboard extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
In CRA (create react app) [plugin-proposal-class-properties](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties) is already installed in babel, so you can define state as a class property. That allows you to avoid overriding the constructor.